crates/aptos-dkg/src/pvss/fiat_shamir.rs

```diff
@@ -7,52 +7,119 @@
+
+    fn random_scalar_from_uniform_bytes(bytes: &[u8]) -> Self {
+        debug_assert_eq!(bytes.len(), 2 * Self::SCALAR_NUM_BYTES);
+        crate::utils::random::random_scalar_from_uniform_bytes(bytes.try_into().unwrap())
+    }
+
+    fn from_u128(v: u128) -> Self {
+        <Self as ff::PrimeField>::from_u128(v)
+    }
+}
 
 #[allow(non_snake_case)]
-pub trait FiatShamirProtocol<T: Transcript> {
-    /// Append a domain separator for the PVSS protocol, consisting of a sharing configuration `sc`,
-    /// which locks in the $t$ out of $n$ threshold.
+#[allow(private_bounds)]
+pub trait PVSS<S: Scalar, T: Transcript>: ScalarProtocol<S> {
+    /// Append a domain separator for the PVSS protocol (in addition to the transcript-level DST used to initialise the FS transcript),
+    /// consisting of a sharing configuration `sc`, which locks in the $t$ out of $n$ threshold.
     fn pvss_domain_sep(&mut self, sc: &ThresholdConfig);
 
     /// Append the public parameters `pp`.
     fn append_public_parameters(&mut self, pp: &T::PublicParameters);
 
     /// Append the signing pub keys.
-    fn append_signing_pub_keys(&mut self, spks: &Vec<T::SigningPubKey>);
+    fn append_signing_pub_keys(&mut self, spks: &[T::SigningPubKey]);
 
     /// Append the encryption keys `eks`.
-    fn append_encryption_keys(&mut self, eks: &Vec<T::EncryptPubKey>);
+    fn append_encryption_keys(&mut self, eks: &[T::EncryptPubKey]);
 
     /// Append the aux data.
-    fn append_auxs<A: Serialize>(&mut self, aux: &Vec<A>);
+    fn append_auxs<A: Serialize>(&mut self, aux: &[A]);
+    fn append_aux<A: Serialize>(&mut self, aux: &A);
+
+    // fn append_share_commitments<A: Serialize>(&mut self, commitments: &[A]); // Not used yet
 
     /// Appends the transcript
     fn append_transcript(&mut self, trx: &T);
 
     /// Returns a random dual-code word check polynomial for the SCRAPE LDT test.
-    fn challenge_dual_code_word_polynomial(&mut self, t: usize, n: usize) -> Vec<Scalar>;
+    fn challenge_dual_code_word_polynomial(&mut self, t: usize, n: usize) -> Vec<S>;
 
     /// Returns one or more scalars `r` useful for doing linear combinations (e.g., combining
     /// pairings in the SCRAPE multipairing check using coefficients $1, r, r^2, r^3, \ldots$
-    fn challenge_linear_combination_scalars(&mut self, num_scalars: usize) -> Vec<Scalar>;
+    /// TODO: maybe rename this / rewrite comment to mention the Schwartz-Zippel lemma? "challenge_for_schwartz_zippel"
+    fn challenge_linear_combination_scalars(&mut self, num_scalars: usize) -> Vec<S>;
+    // fn challenge_linear_combination_scalar(&mut self) -> S { // Not used yet
+    //     self.challenge_linear_combination_scalars(1)[0]
+    // }
+}
+
+pub trait RangeProof<S: Scalar> {
+    fn append_range_proof_sep(&mut self);
+
+    /// Append the claim of a range proof
+    fn append_range_proof_claim(&mut self, claim: &(&range_proof::Commitment, &[G1Projective]));
```

Comment:
The claim (public statement) is $C,\ell,n$. You seem to also be including the $C_j$'s here.